packages/compat/legacy/class.jetpack-sync-settings.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+/**
+ * Class Jetpack_Sync_Settings
+ *
+ * @deprecated Use Automattic\Jetpack\Sync\Settings
+ */
+class Jetpack_Sync_Settings extends Automattic\Jetpack\Sync\Settings {
```

Comment:
You don't need to extend `Automattic\Jetpack\Sync\Settings` in order to delegate static methods.

```php
class Jetpack_Sync_Settings {
	public function __callStatic( $method, $args ) {
		if ( method_exists( 'Automattic\Jetpack\Sync\Settings', $method ) ) {
			_deprecated_function( __METHOD__, 'jetpack-7.5', 'Automattic\Jetpack\Sync\Settings' );
			call_user_func_array( [ 'Automattic\Jetpack\Sync\Settings', $method ], $args );
		}
		throw new Exception( "Method doesn't exist" );
	}
}
```

Reply:
@gravityrail Isn't it safer/less error prone for the stub class to _be_ an instance of the new class as Miguel has it, though, as opposed to only delegating static methods?

Reply:
> Isn't it safer/less error prone for the stub class to be an instance of the new class 

Since we are not invoking an instance method, there is no advantage to subclassing. It just introduces new invisible behaviours into this class unnecessarily.